app/javascript/utilities/http/request.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Generic request with all the default headers required by the application.
+ *
+ * @example
+ * import { request } from '@utilities/http';
+ *
+ * const response = await request('/notification_subscriptions/Article/26')
+ *
+ * Note:
+ * The body option will typically be passed in as a JavaScript object.
+ * A check is performed for this and automatically convert it to JSON if necessary.
+ *
+ * Requests send JSON by default but this can be easily overridden by adding
+ * the Accept and Content-Type headers to the request options.
+ *
+ * The default method is GET.
+ *
+ * @param {string} url The URL to make the request to.
+ * @param {RequestInit} [options={}] The request options.
+ *
+ * @return {Promise<Response>} the response
+ */
+export async function request(url, options = {}) {
+  const {
+    headers,
+    body,
+    method = 'GET',
+    csrfToken = await getCsrfToken(),
```

Comment:
@benhalpern, I found the utility function to retrieve the CSRF token consistently now (including when a page is cached). The request utility now uses `getCsrfToken()`. This should be good to go now.